editor/src/messages/tool/tool_message_handler.rs

```diff
@@ -25,6 +25,7 @@ pub struct ToolMessageHandler {
 	pub tool_state: ToolFsmState,
 	pub transform_layer_handler: TransformLayerMessageHandler,
 	pub shape_editor: ShapeState,
+	pub tool_is_active: bool,
```

Comment:
Is this just the inverse of `document.is_graph_overlay_open()`? If so, could it be used instead to reduce duplicated state?

Reply:
There might be other callers of the activate and deactivate methods in the future (or even now? I didn't look closely how it's currently used). So I don't want to necessarily couple those activate/deactivate methods to the concept of the graph being visible.